### PR TITLE
Let transcripts be processed by personal assistants

### PR DESCRIPTION
**Pull request by albandum**
## Description

For now we were only able to process transcripts through Shared or Company assistants
Some users complained not understanding why they weren't receiving their transcripts.
This PR addresses the issue and lets personal assistants process the transcripts.

- This PR adds Authenticator.fromUserIdAndWorkspaceId

## Deploy Plan

Wait for https://github.com/dust-tt/dust/pull/6222 to be merged, switch the base of this PR to main then merge.

### CODE REVIEW

File: front/temporal/labs/activities.ts [hunk 1]
```diff
@@ -132,14 +132,15 @@ export async function processTranscriptActivity(
...
-  if (!user) {
+  const userRes = await User.findByPk(transcriptsConfiguration.userId);
+  if (!userRes) {
     mainLogger.error(
       { fileId },
       "[processTranscriptActivity] User not found. Stopping."
     );
     return;
   }
+  const user = renderUserType(userRes);
```
Comment (lasryaric) [inline front/temporal/labs/activities.ts:143]: You can use `unsafeGetUsersByModelId` from `front/lib/api/user.ts`

File: front/temporal/labs/activities.ts [hunk 2]
```diff
@@ -239,78 +228,105 @@ export async function processTranscriptActivity(
     return;
   }
 
-  const agentConfigurationsRes = await dustAPI.getAgentConfigurations();
-  if (agentConfigurationsRes.isErr()) {
-    return new Err(new Error(agentConfigurationsRes.error.message));
+  const agent = await AgentConfiguration.findOne({
```
Comment (lasryaric) [inline front/temporal/labs/activities.ts:231]: use `getAgentConfiguration()` from `front/lib/api/assistant/configuration.ts`

**Review by lasryaric — CHANGES_REQUESTED** [review]
Add a `Authenticator::fromUserIdAndWorkspaceId()`

File: front/lib/auth.ts [hunk 3]
```diff
@@ -153,6 +153,58 @@ export class Authenticator {
     });
   }
 
+  /**
+   * Get an Authenticator for the target workspace associated with the specified user.
```
Comment (lasryaric) [inline front/lib/auth.ts:157]: Add a comment saying "to be used only in context where you can't get an authenticator object from a secured key (session or API Key).

**Comment by albandum** [comment]
Replaced by https://github.com/dust-tt/dust/pull/6323 after conflicts resolution of UserResource

**Pull request closed without merging**